pywhy_graphs/algorithms/tests/test_cpdag.py

```diff
@@ -0,0 +1,243 @@
+            (3, 1, {"order": 3}),
+            (1, 2, {"order": 2}),
+            (1, "b", {"order": 1}),
+            (2, "a", {"order": 0}),
+        ]
+        assert set(G.edges.data(data="order")) == set(
+            [(src, target, order["order"]) for src, target, order in expected_order]
+        )
+
+    def test_order_edges_ex1(self):
+        G = nx.DiGraph()
+
+        # 1 -> 3; 1 -> 4; 1 -> 5;
+        # 2 -> 3; 2 -> 4; 2 -> 5;
+        # 3 -> 4; 3 -> 5;
+        # 4 -> 5;
+        G.add_edges_from([(1, 3), (1, 4), (1, 5), (2, 3), (2, 4), (2, 5), (3, 4), (3, 5), (4, 5)])
+        G = order_edges(G)
+        expected_order = [
+            (1, 3, {"order": 7}),
+            (1, 4, {"order": 4}),
+            (1, 5, {"order": 0}),
+            (2, 3, {"order": 8}),
+            (2, 4, {"order": 5}),
+            (2, 5, {"order": 1}),
+            (3, 4, {"order": 6}),
+            (3, 5, {"order": 2}),
+            (4, 5, {"order": 3}),
+        ]
+        assert set(G.edges.data(data="order")) == set(
+            [(src, target, order["order"]) for src, target, order in expected_order]
+        )
+
+
+class TestLabelEdges:
+    def test_label_edges_raises_error_for_non_dag(self):
+        # Test that label_edges raises a ValueError for a non-DAG
+        G = nx.DiGraph([(1, 2), (2, 3), (3, 1)])  # A cyclic graph
+        with pytest.raises(ValueError, match="G must be a directed acyclic graph"):
+            label_edges(G)
+
+    def test_label_edges_raises_error_for_unordered_edges(self):
+        # Test that label_edges raises a ValueError for unordered edges
+        G = nx.DiGraph([(1, 2), (2, 3)])
+        with pytest.raises(
+            ValueError, match="G must have all edges ordered via the `order` attribute"
+        ):
+            label_edges(G)
+
+    def test_label_edges_all_compelled(self):
+        # Create an example DAG for testing
+        G = nx.DiGraph()
+
+        # 1 -> 3; 3 -> 4; 3 -> 5
+        # 2 -> 3;
+        # 4 -> 5
+        G.add_edges_from([(1, 3), (2, 3), (3, 4), (3, 5), (4, 5)])
+        nx.set_edge_attributes(G, None, "order")
+        G = order_edges(G)
+        labeled_graph = label_edges(G)
+
+        expected_labels = {
+            (1, 3): EDGELABELS.COMPELLED,
+            (2, 3): EDGELABELS.COMPELLED,
+            (3, 4): EDGELABELS.COMPELLED,
+            (3, 5): EDGELABELS.COMPELLED,
+            (4, 5): EDGELABELS.REVERSIBLE,
+        }
+        for edge, expected_label in expected_labels.items():
+            assert labeled_graph[edge[0]][edge[1]]["label"] == expected_label, (
+                f"Edge {edge} has label {labeled_graph[edge[0]][edge[1]]['label']}, "
+                f"but expected {expected_label}"
+            )
+
+
+class TestPDAGtoDAG:
+    def test_pdag_to_dag_errors(self):
+        G = nx.DiGraph()
+        G.add_edges_from([("X", "Y"), ("Z", "X")])
+        G.add_edge("A", "Z")
+        G.add_edges_from([("A", "B"), ("B", "A"), ("B", "Z")])
```

Comment:
Any reason why this isn't one add_edges_from call?

Reply:
Good point. Consolidated.